src/middleware/routeHandler.ts

```diff
@@ -0,0 +1,250 @@
+  req: any,
+  res: any,
+  next: any
+) => {
+  routeHandler(req, res).catch((err: any) => {
+    next(err)
+  })
+}
+
+// Used when there are write API calls to the repo on GitHub
+export const attachWriteRouteHandlerWrapper = (routeHandler: any) => async (
+  req: any,
+  res: any,
+  next: any
+) => {
+  const { siteName } = req.params
+  const { growthbook } = req
+
+  let isGitAvailable = true
+
+  // only check git file lock if the repo is ggs enabled
+  if (growthbook?.getFeatureValue(FEATURE_FLAGS.IS_GGS_ENABLED, false)) {
+    isGitAvailable = await handleGitFileLock(siteName, next)
+  }
+
+  if (!isGitAvailable) return
+
+  try {
+    await lock(siteName)
+  } catch (err) {
+    next(err)
+    return
+  }
+
+  await routeHandler(req, res, next).catch(async (err: any) => {
+    await unlock(siteName)
+    next(err)
+  })
+
+  try {
+    await unlock(siteName)
+  } catch (err) {
+    next(err)
+  }
+}
+
+export const attachRollbackRouteHandlerWrapper = (routeHandler: any) => async (
+  req: any,
+  res: any,
+  next: any
+) => {
+  const { userSessionData } = res.locals
+  const { siteName } = req.params
+
+  const { accessToken } = userSessionData
+  const { growthbook } = req
+
+  const shouldUseGitFileSystem = !!growthbook?.getFeatureValue(
+    FEATURE_FLAGS.IS_GGS_ENABLED,
+    false
+  )
+
+  const isGitAvailable = await handleGitFileLock(siteName, next)
+  if (!isGitAvailable) return
+  try {
+    await lock(siteName)
+  } catch (err) {
+    next(err)
+    return
+  }
+
+  let originalStagingCommitSha: any
+  let originalStagingLiteCommitSha: any
+
+  if (shouldUseGitFileSystem) {
+    const results = await Promise.all([
+      gitFileSystemService.getLatestCommitOfBranch(siteName, STAGING_BRANCH),
+      gitFileSystemService.getLatestCommitOfBranch(
+        siteName,
+        STAGING_LITE_BRANCH
+      ),
+    ])
+    const [stagingResult, stagingLiteResult] = results
+
+    if (stagingResult.isErr() || stagingLiteResult.isErr()) {
+      await unlock(siteName)
+      if (stagingResult.isErr()) next(stagingResult.error)
+      if (stagingLiteResult.isErr()) next(stagingLiteResult.error)
+      return
+    }
+    originalStagingCommitSha = stagingResult.value.sha
+    originalStagingLiteCommitSha = stagingLiteResult.value.sha
+    if (!originalStagingCommitSha || !originalStagingLiteCommitSha) {
+      await unlock(siteName)
+      if (stagingResult.isErr()) next(stagingResult.error)
+      return
+    }
+    // Unused for git file system, but to maintain existing structure
+    res.locals.githubSessionData = new GithubSessionData({
+      currentCommitSha: "",
+      treeSha: "",
+    })
+  } else {
+    try {
+      const {
+        currentCommitSha: currentStgCommitSha,
+        treeSha: stgTreeSha,
+      } = await getCommitAndTreeSha(siteName, accessToken, STAGING_BRANCH)
+
+      const {
+        currentCommitSha: currentStgLiteCommitSha,
+      } = await getCommitAndTreeSha(siteName, accessToken, STAGING_LITE_BRANCH)
```

Comment:
would we ever need this as this implies they're not on efs (they're not using git file system)?